ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -161,7 +162,12 @@ public BlockTransactionSelector(
     blockWorldStateUpdater = worldState.updater();
     txWorldStateUpdater = blockWorldStateUpdater.updater();
     blockTxsSelectionMaxTime = miningConfiguration.getBlockTxsSelectionMaxTime();
-    balBuilder = new BlockAccessList.BlockAccessListBuilder();
+
+    maybeBlockAccessListBuilder =
+        protocolSpec
+            .getBlockAccessListFactory()
```

Comment:
Non blocking suggestion, to keep the clean of all the checks about the presence or not of the BAL builder, what if instead having the factory that just return a NoOp BAL builder when it is disabled?

Reply:
No, because for testing reasons for the moment the option can be activated by flag and also by fork. The idea is to be able to test BAL on the current mainnet too

Reply:
@mirgee Do you remember exactly why re used this solution. I forgot the main reason. At first there was a noop but that was changed.

Reply:
@matkt I am not sure I understand the question - what aspect of the solution are we talking about? There never was a noop builder in the first iteration. I think that the code can be refactored to use a noop builder, but I wouldn't say it's the first priority right now.